Functions/functions_collections_home_task_edited.py

```diff
@@ -0,0 +1,81 @@
+"""
+Home task for Module 4 - Functions (Based on Module 2 - Collections home task)
+
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+2. get previously generated list of dicts and create one common dict:
+
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+import random
+import string
+
+
+# Function for generation of single dictionary with defined ranges of random keys and values
+def generate_dictionary_with_random_keys_and_values(
+        set_of_keys,
+        start_of_value, end_of_value,
+        start_of_total_count, end_of_total_count,
+):
+    final_dictionary = {random.choice(set_of_keys): random.randint(start_of_value, end_of_value)
+                        for _ in range(random.randint(start_of_total_count, end_of_total_count))}
+    return final_dictionary
```

Comment:
```suggestion
def generate_random_keys_values_dict(keys, min_value, max_value, min_count, max_count):
    """
    Function generates a dictionary with defined ranges of random keys and values.
    @param keys: possible keys set
    @param min_value: min possible number for value
    @param max_value: max possible number for value
    @param min_count: min count of values to generate
    @param max_count: max count of values to generate
    @return: random key to random value dict with random count of {key: value} pairs
    """
    return {random.choice(keys): random.randint(min_value, max_value) for _ in range(random.randint(min_count, max_count))}
```
It's just an example how funciton may look like. Here I added a docstring (pls read what it's about), renamed function itself, renamed arguments. Try to not use 'of', 'with' and another prepositions. I hope I showed you a good example of naming. Also never use such things as 'final', 'pre_final' bla bla it makes no sense. Try to use such variable names which will makes sense for reader. If your function does only one thing you can to not name variable at all, just return it in place.

Reply:
Tried to make all functions using this code stile.